MatrixKit/Models/Account/MXKAccount.m

```diff
@@ -1673,6 +1677,39 @@ - (void)launchInitialServerSync
     }];
 }
 
+- (void)attemptDeviceDehydration
+{
+    [self attemptDeviceDehydrationWithRetry:YES];
+}
+
+- (void)attemptDeviceDehydrationWithRetry:(BOOL)retry
+{
+    if (![MXKAppSettings standardAppSettings].enableDehydration) {
+        return;
+    }
+    
+    MXKeyData * keyData = [[MXKeyProvider sharedInstance] requestKeyForDataOfType:MXDehydrationServiceKeyDataType isMandatory:NO expectedKeyType:kRawData];
+    if (keyData == nil)
+    {
+        MXLogWarning(@"[MXKAccount] no key provided for device dehydration");
+    }
+    
+    NSData *key = ((MXRawDataKey*) keyData).key;
+    
+    MXLogDebug(@"[MXKAccount] starting device dehydration");
+    [[MXKAccountManager sharedManager].dehydrationService dehydrateDeviceWithMatrixRestClient:mxRestClient crypto:mxSession.crypto dehydrationKey:key success:^(NSString *deviceId) {
+        [MXKAppSettings standardAppSettings].enableDehydration = NO;
+        MXLogDebug(@"[MXKAccount] device successfully dehydrated");
+    } failure:^(NSError *error) {
+        if (retry) {
```

Comment:
Code format
We miss a code format linter in the CI :/

Reply:
Yup, locally as well but I think that ship has sailed for ObjC